src/S3Persistor.js

```diff
@@ -180,6 +184,14 @@ module.exports = class S3Persistor extends AbstractPersistor {
         )
       }
     }
+
+    if (response.IsTruncated) {
+      await this.deleteDirectory(
+        bucketName,
+        key,
+        response.NextContinuationToken
+      )
+    }
```

Comment:
Nitpick: this could be written in a loop to avoid building the call stack:
```js
let continuationToken
while (true) {
  const options = ...
  if (continuationToken) {
    options.ContinuationToken = continuationToken
  }
  response = await s3.listObjectsV2(...)
  if (!response.IsTruncated) {
    break
  }
  continuationToken = response.NextContinuationToken
}
```
but with the call stack being [around 10,000 stacks deep](https://2ality.com/2014/04/call-stack-size.html), it's only a problem with 10M objects, which is probably fine.

Reply:
I'd figured that using `await` effectively drops us out of the call stack at this point anyway, but I'm never quite sure how this stuff works itself out. I guess it's still a concurrent method call to keep track of, stack or no?

Reply:
ah! yes indeed. I also expect that there's some memory taken for keeping track of the recursive calls, but I don't know the details.